packages/jaeger-ui/src/demo/trace-generators.js

```diff
@@ -147,7 +147,7 @@ export default chance.mixin({
       operationName: chance.pickone(operations),
       references: [],
       startTime,
-      duration: chance.integer({ min: 1, max: traceEndTime - startTime }),
+      duration: chance.integer({ min: 1, max: Math.max(1, traceEndTime - startTime)}),
```

Comment:
this is a band-aid that hides the underlying issue. If we have the expectation that trace/span duration should be more than 1 (it's 1 microsecond here, iirc), then the data generator should act accordingly and not generate span durations that are too small

Reply:
And another issue is that even if the actual span durations are not small, it seems like this specific subtraction can produce a small value, which also indicates some weird timing of the generated span.

Reply:
you need to understand why this condition could happen.